File: 1078-remove-outermost-parentheses/solution.java
class Solution {
    public String removeOuterParentheses(String s) {
        String result="";
        int depth=0;
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='('){
                if(depth>0){
                    result=result+s.charAt(i);
                }
                depth++;
            }
            else{
                depth--;
                if(depth>0){
                    result=result+s.charAt(i);
                }
            }
        }
        return result;
    }
}

File: 1078-remove-outermost-parentheses/solution.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        result=""
        depth=0
        for ch in s:
            if ch=='(':
                if(depth>0):
                    result=result+ch
                depth=depth+1
            else:
                depth=depth-1
                if(depth>0):
                    result=result+ch
        return result
